way_to_home/custom_user/views.py

```diff
@@ -116,3 +125,66 @@ def signin_google(request):
         return HttpResponse("User was successfully created", status=201)
 
     return HttpResponse("User's data is empty", status=400)
+
+
+@require_http_methods(["POST"])
+def reset_password(request):
+    """Function that provides reset user password"""
+    data = request.body
+    email = data.get('email')
+    if not email:
+        return RESPONSE_400_OBJECT_NOT_FOUND
+    user = CustomUser.get_by_email(email=email)
+    if not user:
+        return RESPONSE_400_OBJECT_NOT_FOUND
+    token = create_token(data={'email': user.email})
+    send_email_password_update(user, token)
+    return RESPONSE_200_OK
+
+
+@require_http_methods(['PUT'])
+def confirm_reset_password(request, token):
+    """Function that provides confirm reset user password"""
+    data = request.body
+    new_password = data.get('new_password')
+    confirm = decode_token(token)
+
+    if not confirm:
+        return RESPONSE_498_INVALID_TOKEN
+    user = CustomUser.get_by_email(email=confirm.get('email'))
+
+    if not user or not new_password:
+        return RESPONSE_400_OBJECT_NOT_FOUND
+
+    if not password_validator(new_password):
+        return RESPONSE_400_INVALID_DATA
+
+    if not user.check_password(new_password):
+        is_updated = user.update(password=new_password)
+
+        if not is_updated:
+            return RESPONSE_400_DB_OPERATION_FAILED
+
+        send_successful_update_email(user)
+        return RESPONSE_200_UPDATED
+
+    return RESPONSE_400_INVALID_DATA
+
+
+@require_http_methods(["PUT"])
+def change_password(request):
+    """Function that provides change user password"""
+    user = request.user
+    data = request.body
+    new_password = data.get('new_password')
+    old_password = data.get('old_password')
+    if not new_password or not old_password:
+        return RESPONSE_400_OBJECT_NOT_FOUND
+    if user.check_password(old_password):
+        if password_validator(new_password):
+            is_updated = user.update(password=new_password)
+            if not is_updated:
+                return RESPONSE_400_DB_OPERATION_FAILED
+            return RESPONSE_200_UPDATED
+        return RESPONSE_400_INVALID_DATA
+    return RESPONSE_400_INVALID_DATA
```

Comment:
Do not you want to change some code logic?
For example, it can be like this:

```
if not user.check_password(old_password) or not password_validator(new_password):
    return RESPONSE_400_INVALID_DATA
is_updated = user.update(password=new_password)
if not is_updated:
    return RESPONSE_400_DB_OPERATION_FAILED
return RESPONSE_200_UPDATED
```
I think it is more readable :)